Source/WebCore/rendering/MarkedText.h

```diff
@@ -68,24 +69,53 @@ struct MarkedText {
         Frontmost
     };
 
-    unsigned startOffset;
-    unsigned endOffset;
-    Type type;
-    const RenderedDocumentMarker* marker { nullptr };
-    AtomString highlightName { };
+    MarkedText(unsigned startOffset, unsigned endOffset, Type type, const RenderedDocumentMarker* marker = nullptr, AtomString highlightName = { }, int priority = 0) : startOffset(startOffset), endOffset(endOffset), type(type), marker(marker), highlightName(highlightName), priority(priority) { };
+    MarkedText(WTF::HashTableDeletedValueType) : startOffset(std::numeric_limits<unsigned>::max()) { };
+    MarkedText() { };
 
     bool isEmpty() const { return endOffset <= startOffset; }
-    bool operator==(const MarkedText& other) const
-    {
-        return startOffset == other.startOffset && endOffset == other.endOffset && type == other.type && marker == other.marker && highlightName == other.highlightName;
-    }
+    bool isHashTableDeletedValue() const { return startOffset == std::numeric_limits<unsigned>::max(); }
+    bool operator==(const MarkedText& other) const = default;
 
     WEBCORE_EXPORT static Vector<MarkedText> subdivide(const Vector<MarkedText>&, OverlapStrategy = OverlapStrategy::None);
 
     static Vector<MarkedText> collectForDocumentMarkers(const RenderText&, const TextBoxSelectableRange&, PaintPhase);
     static Vector<MarkedText> collectForHighlights(const RenderText&, const TextBoxSelectableRange&, PaintPhase);
     static Vector<MarkedText> collectForDraggedContent(const RenderText&, const TextBoxSelectableRange&);
+
+    unsigned startOffset { 0 };
+    unsigned endOffset { 0 };
+    Type type { Type::Unmarked };
+    const RenderedDocumentMarker* marker { nullptr };
+    AtomString highlightName { };
```

Comment:
Nit: I don't think you need `{ }`. Please check if the compiler is happy without it.

Reply:
the complier is happy!